[docs] Clarify for what version the Helm chart docs are applicable

chaudum opened the pull request:
**What this PR does / why we need it**:

The docs and reference of the Helm chart in the documentation is only applicable for the Loki Helm chart `>= 3.0` and therefore does not apply to the `grafana/loki-stack` chart, which depends on the Loki Helm chart `^2.15.2` ([requirements.yaml](https://github.com/grafana/helm-charts/blob/main/charts/loki-stack/requirements.yaml#L5)).

Also added a headline to the "Install Loki using Helm" index page, because it looks awkward without, see screenshot.

![2023-01-17_20-28-52](https://user-images.githubusercontent.com/281260/212994261-f817f556-4cbf-454f-920f-875c5c085689.png)


**Which issue(s) this PR fixes**:

Fixes: #8152

Signed-off-by: Christian Haudum <christian.haudum@gmail.com>



**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

MichelHollands reviewed (commented):
LGTM but seems like you have to run helm-docs again.

chaudum commented:
> LGTM but seems like you have to run helm-docs again.

I don't know how the helm check passed previously, because I did not change the README. However, I fixed the diversion.

I also added a headline to the index page.

trevorwhitney commented on docs/sources/installation/helm/reference.md:
```diff
@@ -17,6 +17,11 @@ keywords: []
 This is the generated reference for the Loki Helm Chart values.
 
+> **Note:** This reference is for the Loki Helm chart version 3.0 or greater.
+> If you are using the `grafana/loki-stack` Helm chart from the community repo,
```
should we add the other "unsupported" charts not replaced by this chart as well, such as `grafana/loki-distributed`?

trevorwhitney reviewed (approved):
LGTM, thanks!
